lib/system.rb

```diff
@@ -0,0 +1,152 @@
+require 'date'
+
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'hoteldate'
+require_relative 'hotelblock'
+require_relative 'custom_errors'
+
+module Hotel
+  class System
+    
+    attr_reader :rooms, :reservations, :dates, :hotelblocks
+    
+    def initialize
+      @rooms = Hotel::Room.generate_rooms
+      @reservations = []
+      @dates = []
```

Comment:
A comment to tell us what `@dates` is meant to hold would be useful, as it's not immediately obvious.